ministryofjustice/laa-review-criminal-legal-aid
# [CRIMAPP-91] display caseworker competency history

## Description of change
Adds a history link to the manage competency page to enable supervisors to view the history of changes to a caseworkers competencies. If there is no history, the table is not visible and placeholder text is displayed. 

Also moves the link to change competencies from the caseworkers name to the competency list (see screenshots below).

## Link to relevant ticket
[CRIMAPP-91](https://dsdmoj.atlassian.net/browse/CRIMAPP-91)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:
<img width="1310" alt="Screenshot 2023-11-14 at 14 21 08" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/5789df59-3b54-4181-8122-06d52aaf5b1b">


### After changes:
<img width="1310" alt="Screenshot 2023-11-14 at 13 58 43" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/5b6cb771-36fc-46c1-896b-acf850cafc66">
<img width="1310" alt="Screenshot 2023-11-14 at 13 31 10" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/db12acf3-b842-416a-ae5f-d78798f6e65f">

<img width="1310" alt="Screenshot 2023-11-14 at 14 12 53" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/4b775fec-9382-4ba1-bdf1-417af79cbcfb">
<img width="1310" alt="Screenshot 2023-11-14 at 14 13 03" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/24d423c6-c7c7-4356-b28f-6ed155a04b68">

## How to manually test the feature

Requires you to have at least one supervisor and caseworker user.

Log into review as a supervisor. You should see the manage competencies tab and be able to click onto it to view the dashboard. 

There you should be click onto the history link that will take you the history page. 

If you have not set any competencies for your caseworker, then you should not see the history table and in it's place you should see the text 'There is no history'.

Play around with setting/unsetting the competencies of your caseworker and confirm that you can see the correct record of changes in the history page. 


[CRIMAPP-91]: https://dsdmoj.atlassian.net/browse/CRIMAPP-91?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
